lib/berlim/guardian.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Berlim.Guardian do
+  @moduledoc """
+  The Guardian file
+  """
+
+  use Guardian, otp_app: :berlim
+  alias Berlim.InternalAccounts
+
+  def subject_for_token(user, _claims) do
+    {:ok, to_string(user.id)}
+  end
+
+  def resource_from_claims(%{"sub" => id, "type" => type}) do
+    case get_resource(id, type) do
+      nil ->
+        {:error, :resource_not_found}
+
+      user ->
+        {:ok, user}
+    end
+  end
+
+  defp get_resource(id, type) do
+    case type do
+      "Admin" ->
+        InternalAccounts.get_admin!(id)
+
+      "Taxi" ->
+        InternalAccounts.get_taxi!(id)
+
+      _ ->
+        nil
```

Comment:
pra me guardar desse `nil` aqui, eu faria um guard no `resource_from_claims when is_binary(type)`

Reply:
mas o guard não cobre o caso de se for enviado um type diferente de 'Admin' e 'Taxi' 
`(CaseClauseError) no case clause matching: "Xpto"`

Reply:
que tal assim? 
```
 def resource_from_claims(%{"sub" => id, "type" => "Admin"}) do
    user = InternalAccounts.get_admin!(id)
    {:ok, user}
  end

  def resource_from_claims(%{"sub" => id, "type" => "Taxi"}) do
    user = InternalAccounts.get_taxi!(id)
    {:ok, user}
  end

  def resource_from_claims(%{"sub" => _, "type" => _}) do
    {:error, :resource_not_found}
  end
```